kubespawner/spawner.py

```diff
@@ -2673,6 +2702,10 @@ async def progress(self):
         progress = 0
         next_event = 0
 
+        # timers for deciding when to show slow spawn patience message
+        start_time = time.perf_counter()
+        last_message_time = float()
```

Comment:
One last thing:

```suggestion
        last_message_time = start_time
```

then I think we're set!